src/index.js

```diff
@@ -57,6 +59,20 @@ class Toggle extends Component {
   setOff = this.setOnState.bind(this, false)
   toggle = this.setOnState.bind(this, undefined)
 
+  toggleKeys = ['Enter', ' '] // This matches <button> behavior
+  handleKeyUp = event => {
```

Comment:
I'm ok with this implementation, but why doesn't `onClick` fire in addition to `onKeyUp`? I would expect `this.toggle` to be called twice 🤔

Reply:
`onClick` does fire as well in certain cases. That's what I wanted input on. I ran into this, so I check the target in the body of `handleKeyUp`. If it's a `<button>`, we do nothing, since buttons respond to both `Enter` and `Space`. If it's an `<input>`, we don't respond to `Space` because browsers will already treat it as a click. However, we will respond to `Enter` to match `<button>`. For any other elements, we respond to both `Enter` and `Space`.

I'm not a _huge_ fan of this, and I worry that there's a missing corner case. However, that's what I came up with. Can this be improved?

Reply:
What if we make a new prop getter that has the keyup prop. So then the user can decide. Could be: `getNonButtonTogglerProps` or something. I think the implementation would be simpler, though it'd make the API potentially more confusing... Maybe this is good as is... 🤔

Reply:
Actually, I think you have a good point. That brings up another idea that I'll propose.

```JavaScript
getInputTogglerKeyUpProp(onKeyUp) {
  return {
    onKeyUp: (event) => {
      onKeyUp(event)
      if (event.key === ' ') {
        this.toggle()
      }
    }
  }
}
```

```JavaScript
getGenericTogglerKeyUpProp(onKeyUp) {
  return {
    onKeyUp: (event) => {
      onKeyUp(event)
      if (toggleKeys.indexOf(event.key) > -1) {
        this.toggle()
      }
    }
  }
}
```

Usage:

`<input type="checkbox" {...getTogglerProps()} {...getInputTogglerKeyUpProp()}>`

`<span {...getTogglerProps()} {...getGenericTogglerKeyUpProp()}>`

Basically, `getTogglerProps` is designed for `<button>`s. We add these extra methods to support other elements. We'll just be sure to document.

Thoughts?

Reply:
I was thinking that `getInputTogglerKeyUpProp` would actually do:

```javascript
getInputTogglerKeyUpProp(onKeyUp) {
  return this.getTogglerProps({
    onKeyUp: (event) => {
      onKeyUp(event)
      if (event.key === ' ') {
        this.toggle()
      }
    }
  })
}
```

Same with the generic one...